### ISSUE TITLE
Add the ability to manually define an aria-label for the navigation block

### ISSUE THREAD

**Issue by fabiankaegy** [WordPress/gutenberg#66935]
The core navigation block uses the name of the Navigation post that is assigned to it as it's aria label. 

This behavior is not documented/called out anywhere. 

Since the name of a navigation menu in the admin isn't necessarily very prescriptive to a user using a screen reader on the front end of the site, it would be great to be able to decouple this aria-label from the title of the navigation post.

We should still leave that behavior as the default/fallback in case no label gets specified manually. However, having the ability to define a custom label via the advanced settings panel would be a great improvement.

This way that actual label could already get defined in code rather than relying on the user to set it correctly.

**Comment by up1512001** [comment]
@fabiankaegy raised PR to address this, can you please review 🙇‍♂ 
https://github.com/WordPress/gutenberg/pull/66943
